wm/client.go

```diff
@@ -0,0 +1,258 @@
+		if proto == "WM_DELETE_WINDOW" {
+			askNicely = true
+		}
+	}
+
+	if !askNicely {
+		err := xproto.DestroyWindowChecked(cli.wm.x.Conn(), cli.win).Check()
+		if err != nil {
+			fyne.LogError("Close Error", err)
+		}
+
+		return
+	}
+
+	protocols, err := xprop.Atm(cli.wm.x, "WM_PROTOCOLS")
+	if err != nil {
+		fyne.LogError("Get Protocols Error", err)
+		return
+	}
+
+	delWin, err := xprop.Atm(cli.wm.x, "WM_DELETE_WINDOW")
+	if err != nil {
+		fyne.LogError("Get Delete Window Error", err)
+		return
+	}
+	cm, err := xevent.NewClientMessage(32, cli.win, protocols,
+		int(delWin))
+	err = xproto.SendEventChecked(cli.wm.x.Conn(), false, cli.win, 0,
+		string(cm.Bytes())).Check()
+	if err != nil {
+		fyne.LogError("Window Delete Error", err)
+	}
+}
+
+func (cli *client) iconifyMessage(state int) {
+	iconifyAtm, err := xprop.Atm(cli.wm.x, "WM_STATE_CHANGE")
+	if err != nil {
+		fyne.LogError("Error getting X Atom", err)
+		return
+	}
+	cm, err := xevent.NewClientMessage(32, cli.win, iconifyAtm,
+		state)
+	if err != nil {
+		fyne.LogError("Error creating client message", err)
+		return
+	}
+	err = xevent.SendRootEvent(cli.wm.x, cm, xproto.EventMaskSubstructureNotify|xproto.EventMaskSubstructureRedirect)
+}
+
+func (cli *client) Iconify() {
+	cli.iconifyMessage(icccm.StateIconic)
+	cli.iconic = true
+}
+
+func (cli *client) Uniconify() {
+	cli.iconifyMessage(icccm.StateNormal)
+	cli.iconic = false
+}
+
+func (cli *client) maximizeMessage(action int) {
+	ewmh.WmStateReqExtra(cli.wm.x, cli.win, action, "_NET_WM_STATE_MAXIMIZED_VERT",
+		"_NET_WM_STATE_MAXIMIZED_HORZ", 1)
+}
+
+func (cli *client) Maximize() {
+	cli.maximized = true
+	//1 is for adding a new state
+	cli.maximizeMessage(1)
+}
+
+func (cli *client) Unmaximize() {
+	cli.maximized = false
+	//0 is for removing a state
+	cli.maximizeMessage(0)
+}
+
+func (cli *client) Focus() {
+	xproto.SetInputFocus(cli.wm.x.Conn(), 0, cli.win, 0)
+}
+
+func (cli *client) RaiseToTop() {
+	cli.wm.RaiseToTop(cli)
+}
+
+func (cli *client) RaiseAbove(win desktop.Window) {
+	topID := cli.wm.rootID
+	if win != nil {
+		topID = win.(*client).id
+	}
+
+	cli.Focus()
+	if cli.id == topID {
+		return
+	}
+
+	cli.wm.raiseWinAboveID(cli.id, topID)
+}
+
+func (x *x11WM) raiseWinAboveID(win, top xproto.Window) {
+	err := xproto.ConfigureWindowChecked(x.x.Conn(), win, xproto.ConfigWindowSibling|xproto.ConfigWindowStackMode,
+		[]uint32{uint32(top), uint32(xproto.StackModeAbove)}).Check()
+	if err != nil {
+		fyne.LogError("Restack Error", err)
+	}
+}
+
+func (cli *client) addStateHint(hint string) {
+	cli.stateHints = append(cli.stateHints, hint)
+	ewmh.WmStateSet(cli.wm.x, cli.win, cli.stateHints)
+}
+
+func (cli *client) removeStateHint(hint string) {
+	for i, curHint := range cli.stateHints {
+		if curHint == hint {
+			cli.stateHints = append(cli.stateHints[:i], cli.stateHints[i+1:]...)
+			ewmh.WmStateSet(cli.wm.x, cli.win, cli.stateHints)
+			return
+		}
+	}
+}
+
+func (cli *client) maximizeFrame() {
```

Comment:
These two methods are just proxying to the frame, but we are in the same package - it seems unneeded